// swagger-axios-codegen@0.0.1/dist/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GENERIC_SPLIT_KEY = '[';
// 是否是接口类型
exports.isGenerics = (s) => /^.+\[.+\]$/.test(s);
/**
 * 分解泛型接口
 * @param definitionClassName
 */
function getGenericeClassNames(definitionClassName) {
    const splitIndex = definitionClassName.indexOf(exports.GENERIC_SPLIT_KEY);
    // 泛型基类 PagedResultDto
    const interfaceClassName = definitionClassName.slice(0, splitIndex);
    // 泛型类型 T 的类型名称
    const TClassName = definitionClassName.slice(splitIndex + 1, -1);
    return { interfaceClassName, TClassName };
}
exports.getGenericeClassNames = getGenericeClassNames;
/**
 * 获取引用类型
 * @param s
 */
function refClassName(s) {
    let propType = s.slice(s.lastIndexOf('/') + 1);
    if (exports.isGenerics(propType)) {
        const { interfaceClassName, TClassName } = getGenericeClassNames(propType);
        // return `${interfaceClassName}<${toBaseType(TClassName)}>`
        return propType.replace(/\[/g, '').replace(/\]/g, '').replace(/,/g, '');
    }
    else {
        return propType;
    }
}
exports.refClassName = refClassName;
function toBaseType(s) {
    let result = '';
    switch (s) {
        case 'array':
            result = '[]';
            break;
        case 'Int64':
        case 'integer':
            result = 'number';
            break;
        case 'Guid':
        case 'String':
        case 'string':
            result = 'string';
            break;
        case 'file':
            result = 'any';
            break;
        default:
            result = s;
            break;
    }
    return result;
}
exports.toBaseType = toBaseType;
function getMethodName(path) {
    const paths = path.split('/');
    for (let i = paths.length - 1; i >= 0; i--) {
        if (/\{.+\}/.test(paths[i]) === false) {
            return paths[i];
        }
    }
    return '';
}
exports.getMethodName = getMethodName;
